src/app/(sidebar)/transaction/cli-sign/page.tsx

```diff
@@ -0,0 +1,18 @@
+"use client";
+
+import { Routes } from "@/constants/routes";
+import { useStore } from "@/store/useStore";
+import { Loader } from "@stellar/design-system";
+import { useRouter } from "next/navigation";
+import { useEffect } from "react";
+
+export default function CliSignTransaction() {
+  const { network, transaction } = useStore();
+  const { sign } = transaction;
+  const router = useRouter();
+  useEffect(() => {
+    router.push(Routes.SIGN_TRANSACTION);
+  }, [sign.importXdr, network, router]);
+
+  return <Loader />;
+}
```

Comment:
I like this idea of using the route to do the redirect. :raised_hands: Do you think we could move `NetworkByPasswordProvider` logic here and remove it completely from `src/app/layout.tsx`. That way, everything related to CLI sign transaction would be handled in one place.

Reply:
Ah, great idea! I'll get an update pushed shortly